semgrep-core/src/parsing/tree_sitter/Parse_go_tree_sitter.ml

```diff
@@ -51,8 +51,18 @@ let rev = false
    to another type of tree.
 *)
 
-(* TODO: need remove enclosing `` *)
-let raw_string_literal env tok = str env tok
+(* TODO: Update grammar so that the leading and trailing backticks are tokenized
+ * separately, the way interpreted string literals are:
+ * https://github.com/tree-sitter/tree-sitter-go/blob/0fa917a7022d1cd2e9b779a6a8fc5dc7fad69c75/grammar.js#L839-L843
+ * *)
+let raw_string_literal env tok =
+  let (loc, s) = tok in
+  (* Remove leading and trailing backticks. The grammar guarantees that raw
```

Comment:
Probably better to code defensively and add the check, since this depends on behavior of an external library

Reply:
Fair enough, will do. I'll probably add a test that exercises an empty raw string and do a bounds-check with a `failwith`. That way any update that changes this behavior will get us a test failure. Sound good?